scripts/ae_generate_streamlines.py

```diff
@@ -0,0 +1,532 @@
+    )
+
+    parser.add_argument(
+        "--max_total_sampling",
+        help="Config file [ *.json ] for the per-bundle maximum number of generated streamlines."
+        "Use this argument to limit the time spent on RS for hard bundles.",
+    )
+
+    parser.add_argument(
+        "-r",
+        "--ratio",
+        help="Config file [ *.json ] for the per-bundle AE latent space desired seed ratio "
+        "( subject bundle|atlas bundle )",
+    )
+
+    parser.add_argument(
+        "-b",
+        "--bandwidth",
+        help="Bandwidth size used for the parzen window. Default is none, meaning that we used the "
+        "Silverman’s rule of thumb for the bandwidth estimation",
+        type=float,
+        default=None,
+    )
+    parser.add_argument(
+        "-m",
+        "--max_seeds",
+        help="Maximum number of seeds to use per bundle for the AE latent space",
+        type=int,
+    )
+    parser.add_argument(
+        "--minL",
+        help="Minimum length of generated streamlines, in mm.",
+        type=int,
+        default=20,
+    )
+    parser.add_argument(
+        "--maxL",
+        help="Maximum length of generated streamlines, in mm.",
+        type=int,
+        default=220,
+    )
+    parser.add_argument("-p", "--plot_umaps", help="Plot umaps", action="store_true")
+    parser.add_argument(
+        "-a",
+        "--generate_all_bundles",
+        help="If flagged, will generate all bundles present in the atlas, "
+        "even if they were not present in the --in_bundles_common_space argument."
+        "Seed from missing bundle, will entirely be taken from the atlas even if the ratio is not ( 0 | 1 )",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--use_rs",
+        help="If not flagged, will use the default sklearn gaussian sampling process. ",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--batch_sampling",
+        help="Sampling batch size. Use this argument if you want to prevent excessive memory usage. ",
+        type=int,
+        default=5000,
+    )
+    parser.add_argument(
+        "--gmm_n_component",
+        help="Number of components for the Gaussian Mixture Model used as a proposal "
+        "distribution for RS. Only used if --use_rs is flagged.",
+        type=int,
+        default=11,
+    )
+    parser.add_argument(
+        "--degree",
+        help="Config file [ *.json ] for the per-bundle maximum degree angle for the peaks filtering",
+    )
+    parser.add_argument(
+        "--in_transfo",
+        help="Path of the file containing the 4x4 transformation, matrix (.txt, .npy or .mat).",
+    )
+    parser.add_argument(
+        "--in_deformation",
+        help="Path to the file containing a deformation field [ *.nii/.nii.gz ].",
+    )
+
+    parser.add_argument(
+        "--white_matter_config",
+        help="Config file [ *.json ] used to determine the type of WM filtering ( WM mask | thresholded FA ).",
+    )
```

Comment:
So the paradigm of the parameters here is different wrt to e.g. the threshold finding script, where a single config file is used. We will eventually need to be consistent with this.

Reply:
Yep, I guess we could pass config file for the `ae_find_thresholds.py` script. This was mandatory for the `ae_generate_streamlines.py` script as fsl WM mask almost always output 0 value around the fornix/AC/PC regions

Reply:
Maybe to do in another PR ?